x-pack/test/fleet_api_integration/apis/epm/file.ts

```diff
@@ -14,79 +16,126 @@ export default function ({ getService }: FtrProviderContext) {
+            )
+            .set('kbn-xsrf', 'xxx')
+            .expect('Content-Type', 'application/json; charset=utf-8')
+            .expect(200);
+        } else {
+          warnAndSkipTest(this, log);
+        }
+      });
 
-    it('fetches a .json kibana dashboard file', async function () {
-      if (server.enabled) {
-        await supertest
-          .get('/api/fleet/epm/packages/filetest/0.1.0/kibana/dashboard/sample_dashboard.json')
-          .set('kbn-xsrf', 'xxx')
-          .expect('Content-Type', 'application/json; charset=utf-8')
-          .expect(200);
-      } else {
-        warnAndSkipTest(this, log);
-      }
-    });
+      it('fetches a .json kibana dashboard file', async function () {
+        if (server.enabled) {
+          await supertest
+            .get('/api/fleet/epm/packages/filetest/0.1.0/kibana/dashboard/sample_dashboard.json')
+            .set('kbn-xsrf', 'xxx')
+            .expect('Content-Type', 'application/json; charset=utf-8')
+            .expect(200);
+        } else {
+          warnAndSkipTest(this, log);
+        }
+      });
 
-    it('fetches a .json search file', async function () {
-      if (server.enabled) {
+      it('fetches a .json search file', async function () {
+        if (server.enabled) {
+          await supertest
+            .get('/api/fleet/epm/packages/filetest/0.1.0/kibana/search/sample_search.json')
+            .set('kbn-xsrf', 'xxx')
+            .expect('Content-Type', 'application/json; charset=utf-8')
+            .expect(200);
+        } else {
+          warnAndSkipTest(this, log);
+        }
+      });
+    });
+    describe('it gets files from an uploaded package', () => {
+      before(async () => {
+        if (!server.enabled) return;
+        const testPkgArchiveTgz = path.join(
+          path.dirname(__filename),
+          '../fixtures/direct_upload_packages/apache_0.1.4.tar.gz'
+        );
+        const buf = fs.readFileSync(testPkgArchiveTgz);
         await supertest
-          .get('/api/fleet/epm/packages/filetest/0.1.0/kibana/search/sample_search.json')
-          .set('kbn-xsrf', 'xxx')
-          .expect('Content-Type', 'application/json; charset=utf-8')
+          .post(`/api/fleet/epm/packages`)
+          .set('kbn-xsrf', 'xxxx')
+          .type('application/gzip')
+          .send(buf)
           .expect(200);
-      } else {
-        warnAndSkipTest(this, log);
-      }
+      });
+      after(async () => {
+        if (!server.enabled) return;
+        await supertest.delete(`/api/fleet/epm/packages/apache-0.1.4`).set('kbn-xsrf', 'xxxx');
+      });
+      it('fetches a .png screenshot image', async function () {
+        if (server.enabled) {
+          await supertest
+            .get('/api/fleet/epm/packages/apache/0.1.4/img/kibana-apache-new.png')
+            .set('kbn-xsrf', 'xxx')
+            .expect('Content-Type', 'image/png')
+            .expect(200);
```

Comment:
Maybe a test to ensure we got a buffer/binary value? Just something to make sure that the response body is also correct; not just the content type header